examples/async_tasks/external_source_external_thread.rs

```diff
@@ -0,0 +1,83 @@
+use bevy::prelude::*;
+// Using crossbeam_channel instead of std as std `Receiver` is `!Sync`
+use crossbeam_channel::{bounded, Receiver};
+use rand::Rng;
+
+fn main() {
+    let (tx, rx) = bounded::<u32>(10);
```

Comment:
I think doing "app setup" logic like this should be done in startup systems. This is running "timer logic", which shouldn't start until the App itself starts. This will become even more important if we make plugin / resource init deferred.